### PR TITLE
feat(ia): Settings / Connections - APIs

### PR DESCRIPTION
**Pull request by jaredrethman**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

* Migrated from _/wp-admin/admin.php?page=newspack-connections-wizard_.
* Adds Google OAuth and Mailchimp components to APIs section.
* Refactored API Fetch to use new `useWizardApiFetch` hook, introduced in #3163.
* Refactored `ActionCard` components to use `WizardPluginCard`.
* Google OAuth: modernized `getURLParams` function and eliminated the need for the `qs` package.
* Replaced deprecated conditional props on  `<Button />` with preferred `variant` prop.
* Updated `/oauth/mailchimp` to have consistent return type i.e. `{}` not `{}|[]`. 

### How to test the changes in this Pull Request:

**Original location:**
* Connections Wizard: _/wp-admin/admin.php?page=newspack-connections-wizard_

1. Checkout `feat/ia-settings-connections__apis` and build assets.
2. Navigate to _/wp-admin/admin.php?page=newspack-settings_
3. For the below APIs confirm you see one (or both, requires Connect/Disconnect)

#### Google OAuth:

**_NOTE:_**
_I was unable to authenticate with Google OAuth (due to closed-source dependencies). I circumvented all server-side conditionals/guardrails to test._

**Unauthenticated:**
Screenshot 2024-06-05 at 16 21 53

**Authenticated:**
Screenshot 2024-06-05 at 16 15 09

#### Mailchimp:

**Unauthenticated:**
Screenshot 2024-06-05 at 16 25 01

**Authenticated:**
Screenshot 2024-06-05 at 16 24 08

**Additional steps:**
- Confirm data (including errors) are persistent between tab changes.
- Confirm Disconnect & Connect call to actions work as intended.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: assets/wizards/newspack/views/settings/connections/google-oauth.tsx [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+
+/**
+ * Internal dependencies.
+ */
+import { Button } from '../../../../../components/src';
+import WizardsActionCard from '../../../../wizards-action-card';
+import { useWizardApiFetch } from '../../../../hooks/use-wizard-api-fetch';
+import { WizardError, WIZARD_ERROR_MESSAGES } from '../../../../errors';
+
+function getURLParams() {
```
Comment (chickenn00dle) [inline assets/wizards/newspack/views/settings/connections/google-oauth.tsx:19]: Non-blocking question: Can we use react routers [useSearchParams](https://reactrouter.com/en/main/hooks/use-search-params) or [useParams](https://reactrouter.com/en/main/hooks/use-params) here?

File: assets/wizards/newspack/views/settings/connections/google-oauth.tsx [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+		);
+		wizardApiFetch< string >(
+			{
+				path: '/newspack/v1/oauth/google/start',
+			},
+			{
+				onSuccess( url ) {
+					/** authWindow can be 'null' due to browser's popup blocker. */
+					if ( authWindow ) {
+						authWindow.location = url;
```
Comment (chickenn00dle) [inline assets/wizards/newspack/views/settings/connections/google-oauth.tsx]: On Firefox, the auth window opens but the location is never updated, and so I end up with a popup of the homepage:

Screenshot 2024-06-24 at 16 04 39

We should probably not trigger the opening of the popup until the api fetch is successful and we are certain there are no popup blocker issues, etc.

**Review by chickenn00dle — CHANGES_REQUESTED** [review]
Another one looking good @jaredrethman! 

Left one bit of feedback about the auth window then this should be good to go.

File: assets/wizards/newspack/views/settings/connections/google-oauth.tsx (hunk 1 above)
Comment (jaredrethman) [reply assets/wizards/newspack/views/settings/connections/google-oauth.tsx:19]: Thanks for the suggestion @chickenn00dle !

- `useSearchParams` is only available in react-router-dom v6, and we are currently using v5.
- `useParams` handles path parameters, not query parameters.

Given our current setup, using a custom function like getURLParams is straightforward and avoids adding unnecessary complexity or dependencies.

File: assets/wizards/newspack/views/settings/connections/google-oauth.tsx [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+	function openAuth() {
+		let authWindow: Window | null = null;
+		wizardApiFetch< string >(
+			{
+				path: '/newspack/v1/oauth/google/start',
+			},
+			{
+				onSuccess( url ) {
+					authWindow = window.open(
+						'about:blank',
```
Comment (chickenn00dle) [inline assets/wizards/newspack/views/settings/connections/google-oauth.tsx]: We should be able to just open with `url` directly here. I don't think there is a reason to wait until the popup opens to update the location, is there?

File: assets/wizards/newspack/views/settings/connections/google-oauth.tsx (hunk 2 above)
Comment (chickenn00dle) [reply assets/wizards/newspack/views/settings/connections/google-oauth.tsx]: Thanks @jaredrethman! 

This is working as expected now. It looks like there was an issue with my local site config where the `NEWSPACK_GOOGLE_OAUTH_PROXY_OVERRIDE` was not correct which was causing the popup location not to update.

This did get me thinking though, right now if the popup fails to open and `authWindow` is null we simply abort with no additional error message. We should probably handle this case by notifying the publisher that they should disable any popup blocker, etc. and try again.

WDYT?

**Review by chickenn00dle — CHANGES_REQUESTED** [review]

File: assets/wizards/newspack/views/settings/connections/google-oauth.tsx (hunk 2 above)
Comment (jaredrethman) [reply assets/wizards/newspack/views/settings/connections/google-oauth.tsx]: @chickenn00dle great idea, I have gone ahead and added some error reporting for null `authWindow` and `url` values.

File: assets/wizards/newspack/views/settings/connections/google-oauth.tsx (hunk 3 above)
Comment (jaredrethman) [reply assets/wizards/newspack/views/settings/connections/google-oauth.tsx]: @chickenn00dle have updated.

**Review by chickenn00dle — APPROVED** [review]
Nice! Thanks for the changes @jaredrethman! Looks good 👍

**Pull request merged**